src/targets/mobile/lib/withPersistentState.js

```diff
@@ -0,0 +1,20 @@
+import localforage from 'localforage'
+
+// The withPersistentState HOC automatically saves the wrappped components state and restores it when the component is mounted again.
+// Since the saved state will not be available immediately when the wrapped component is mounted, a `componentStateRestored` hook will be called once the state is restored
+export const withPersistentState = (WrappedComponent, persistenceKey) => class extends WrappedComponent {
+  async componentWillMount () {
+    if (super.componentWillMount) super.componentWillMount()
+    const persistedState = await localforage.getItem(persistenceKey)
+    this.setState(prevState => (persistedState || prevState), () => {
+      if (super.componentStateRestored) super.componentStateRestored()
+    })
```

Comment:
I believe it is *generally* recommended to use `componentDidUpdate` because *generally*, you want to do things regardless of why and how the component updated.
However in this case, we want to call the `componentStateRestored` hook, and we want to call it only once, specifically after the state has been restored from local storage. If we used a generic `componentDidUpdate` instead, we would need to add some detection code to make sure the component updated because of the restoration we just did and not something else, which is going to be messy. I'm fairly sure this is a valid use case for the second parameter.